src/lib.rs

```diff
@@ -226,24 +226,199 @@ impl WebexEventStream {
     }
 }
 
+enum Authorization<'a> {
+    None,
+    Bearer(&'a str),
+    Basic {
+        username: &'a str,
+        password: &'a str,
+    },
+}
+type BearerToken = String;
+/// Implements low level REST requests to be used internally by the library
+#[derive(Clone)]
+struct RestClient {
+    host_prefix: HashMap<String, String>,
+    web_client: WebClient,
+}
+
+struct RequestBody<T>
+where
+    Body: From<T>,
+{
+    media_type: &'static str,
+    content: T,
+}
+
+impl RestClient {
```

Comment:
The reason why these operations were moved outside `Webex` is because we need more control over the Authorization and Content-Type headers.